abdiff/config.py

```diff
@@ -31,6 +33,18 @@ def webapp_port(self) -> int:
         port = self.WEBAPP_PORT or "5000"
         return int(port)
 
+    @property
+    def transmogrifier_max_workers(self) -> int:
+        """Maximum number of Transmogrifier containers to run in parallel."""
+        max_workers = self.TRANSMOGRIFIER_MAX_WORKERS or 6
```

Comment:
It's a good call out, and I had struggled with that too.  Roping in [your comment below](https://github.com/MITLibraries/transmogrifier-ab-diff/pull/37#discussion_r1815276649) as well here.

I think it's gets difficult when you consider them together. Here is an option with some renaming as suggested:

```
TRANSMOGRIFIER_MAX_CONTAINERS
TRANSMOGRIFIER_CONTAINER_TIMEOUT
```
- I think they are getting kind of long for env vars...
- I think it's awkward to have singular / plural of "container" in them
- possible to interpret `_MAX_CONTAINERS` as the _total_ number of Transmog instances that can run, not specifically in _parallel_; where I think "workers" tends to imply that

What about the following?

```
TRANSMOGRIFIER_CONCURRENCY
TRANSMOGRIFIER_TIMEOUT
```
- we abandon "container" altogether, as perhaps the configurer / developer doesn't really need to think about _how_ they are getting run, more just how many concurrent "instances" of Transmog are allowed and how long they can run
- prefix of "TRANSMOGRIFIER" indicates they are related, and single word afterwards is easy to parse
- this would trickle through to the `Config` properties as well, e.g. `transmogrifier_concurrency` and `transmogrifier_timeout`

Reply:
@ehanson8 - I've pushed a commit with these proposed changes, https://github.com/MITLibraries/transmogrifier-ab-diff/pull/37/commits/4be03687684cd6e14a0901d93aee7a3d406ba77b.  Let me know if you you think it still needs work!

Reply:
Looks good to me, my approval stands!

Reply:
Hmmmm, curious how both of you feel after our book club discussion where we kinda' decided to distinguish between [concurrency vs. parallelism](https://dynamogeeks.com/wp-content/uploads/2024/08/Concurrency-vs-Parallelism.gif) (i.e., trying not to use "concurrent" and "parallel" interchangeably). 🤔 

That aside, I think I'm leaning towards something like...
```
MAX_PARALLEL_CONTAINERS / CONTAINER_LIMIT
CONTAINER_TIMEOUT
```

Reply:
I'm glad we had that discussion yesterday, and I agree that being intentional about "concurrent" vs "parallel" will be beneficial.  

This is partially why I think arguments like `num_workers` or `max_workers` are pretty standard in APIs that deal with concurrency **and/or** parallelism.  Despite the fact these terms _do_ mean different things, they often exist in a space where a simple toggle here or there can change which one is used, and it would be tedious to update the higher configurations of, "this is how many things I want running at once."

Additionally, I would like to keep `"TRANSMOGRIFIER_"` in the mix, to be explicit we're talking about running multiple instances of Transmog.  Who know, perhaps we'll fire up other "containers" in this ABDiff program, for another reason later.  

With all that in mind.... what about the following?

- `TRANSMOGRIFIER_WORKERS`
  - intentionally generic about threads vs processes, concurrency vs parallelism
  - avoids over-complicating that we invoke Transmog via Docker containers
  - communicates, "how many Transmog instances do you want running at once?"

- `TRANSMOGRIFIER_TIMEOUT`
  - similar story to above
  - communicates, "how long max do you want Transmog to run?"

Reply:
@jonavellecuerdo, @ehanson8 - please see commit https://github.com/MITLibraries/transmogrifier-ab-diff/pull/37/commits/73f806ab5101ea26aedc53cfdd8d6b25f4a8ee3d for this update. 

@ehanson8, given your original comments about this being optional, and I've basically reverted to the original env vars, I will _not_ request a re-review.  But please pushback if this feels off to you.

Reply:
Correct, it was optional so my approval continues to stand!

Reply:
@ghukill Thank you for sharing your thought processes! I like the revert. I think my comment did stem from seeing that the language of the names we landed on "leans into norms and conventions in concurrent/parallel APIs that use 'workers'".

Reply:
@ehanson8, @jonavellecuerdo - I must say, though we ended up where we started in code, it has been a timely and edifying discussion.  Despite being clear headed on I/O vs CPU bounds, threads vs processes, etc., I am absolutely guilty of using "concurrency" and "parallelism" interchangeably in the past, but no longer!  I'm glad that, going forward, we have a shared understanding of what these words imply.